File: apps/server/src/modules/reports/reports.controller.ts
import type { Request, Response } from 'express';
import { reportsService } from './reports.service.js';
import { asyncHandler } from '../../utils/asyncHandler.js';
import { ok } from '../../utils/ApiResponse.js';

export const reportsController = {
  orders: asyncHandler(async (req: Request, res: Response) => {
    const data = await reportsService.ordersReport(req.query.from as string, req.query.to as string);
    return ok(res, data, 'Orders report');
  }),

  deliveries: asyncHandler(async (req: Request, res: Response) => {
    const data = await reportsService.deliveriesReport(req.query.from as string, req.query.to as string);
    return ok(res, data, 'Deliveries report');
  }),
};

File: apps/guzo-api/src/main/java/et/guzo/web/controller/ReportsController.java
package et.guzo.web.controller;

import et.guzo.common.ApiResponse;
import et.guzo.service.ReportsService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/reports")
@RequiredArgsConstructor
public class ReportsController {

    private final ReportsService reportsService;

    @GetMapping("/orders")
    public ApiResponse<Map<String, Object>> orders(
        @RequestParam(required = false) String from,
        @RequestParam(required = false) String to
    ) {
        return ApiResponse.ok(reportsService.ordersReport(from, to));
    }

    @GetMapping("/deliveries")
    public ApiResponse<Map<String, Object>> deliveries(
        @RequestParam(required = false) String from,
        @RequestParam(required = false) String to
    ) {
        return ApiResponse.ok(reportsService.deliveriesReport(from, to));
    }
}
